### PR TITLE
[FIX] website_sale: require a cart before submitting extra info

### PR DESCRIPTION
**Pull request by mobe-odoo**
When user tries to enter extra info from Configure Form from 'settings' so 
'website.sale_get_order' method does not get any order.

Steps to reproduce:
1) Install website_sale module.
2) Click on settings > activate 'Extra Step During Checkout' > click on 'Save' 
   button.
3) Now in 'settings', click on 'Configure Form' under 
  'Extra Step During Checkout'.
4) Now, click on 'Discard' button on top right of screen.
5) Enter something in textbox 'Give us your feedback' > click on 'Next' button 
   and traceback will be generated in backend. 

```
ValueError: not enough values to unpack (expected 1, got 0)
  File "odoo/models.py", line 5379, in ensure_one
    _id, = self._ids
ValueError: Expected singleton: sale.order()
  File "odoo/http.py", line 2114, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1699, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "odoo/service/model.py", line 133, in retrying
    result = func()
  File "odoo/http.py", line 1726, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 1840, in dispatch
    return self.request.registry['ir.http']._dispatch(endpoint)
  File "addons/website/models/ir_http.py", line 234, in _dispatch
    response = super()._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 190, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 716, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/website_sale/controllers/main.py", line 117, in website_form_saleorder
    order._message_log(
  File "addons/mail/models/mail_thread.py", line 2589, in _message_log
    self.ensure_one()
  File "odoo/models.py", line 5382, in ensure_one
    raise ValueError("Expected singleton: %s" % self)

```
Error: ValueError: Expected singleton: sale.order() traceback will be generated
because in method '_message_log' it is expecting a record.

Sentry-4126876202

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mobe-odoo** [comment]
Hello @odoo/sentry_reviewers 
can you please check this PR?

File: addons/website_sale/controllers/main.py [hunk 1]
```diff
@@ -110,6 +110,9 @@ def website_form_saleorder(self, **kwargs):
             return json.dumps({'error_fields': e.args[0]})
 
         order = request.website.sale_get_order()
+        if not order:
+            return json.dumps({'error': "No order found, please add something in cart"})
```
Comment (AntoineVDV) [inline addons/website_sale/controllers/main.py:114]:
```suggestion
            return json.dumps({'error': "No order found; please add a product to your cart."})
```

**Review by AntoineVDV — APPROVED** [review]
@robodoo r+

The commit title is misleading, though. It should rather be something like `[FIX] website_sale: require a cart before submitting extra info"`

**Comment by AntoineVDV** [comment]
@robodoo r-
@robodoo delegate+

**Comment by mobe-odoo** [comment]
@robodoo r+

**Comment by mobe-odoo** [comment]
@robodoo r+

**Pull request closed without merging**